### DISCUSSION TITLE
Unable to build `cargo_without_cmake` example

### DISCUSSION THREAD

**Discussion by dphaldes** [KDAB/cxx-qt#847] [Q&A] [answered]
I copied over the example and when I do cargo run, it errors out (which is not very clear).
the compilation errors happen specifically when building cxx-qt-lib.  [log](https://hastebin.com/share/iseniqepud.rust) 

also it compiles cxx-qt crates twice (maybe not the problem and just related to build.rs)

**Comment by LeonMatthesKDAB** [comment, marked as answer]
Hi @mystchonky ,
thanks for taking the time to report this issue.

This was a bit of a headscracher :sweat_smile: 
I suspect you're using a Beta version of Qt 6.7?

As that seems to have added the overload to QDateTime (see: https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/time/qdatetime.h )
For now, if you can, try to build with Qt 6.6 or 6.5 LTS, that should resolve the issue.

We'll have to sort this out to keep compatibility with 6.7 though.
@Montel any idea how to approach this? :thinking: 
I think the easiest way to go about this is to introduce a `qdatetime_settimezone` method on the C++ side that just calls `datetime.setTimeZone()` as that should behave correctly with the default-argument in both 6.6 and 6.7.

If we supported #765 and knew the Qt minor version, we could also go that route.

**Comment by dphaldes** [comment]
Yeah my bad. I am running the testing repo on arch for KDE plasma RCs and forgot that it might break stuff (with QT 6.7 beta). Let me try older versions :)

**Reply by LeonMatthesKDAB** [reply]
It would have been good if you had mentioned it :sweat_smile: 

In any case, thank you for reporting this!
It's good to have this heads up, so we're prepared once 6.7 is released as stable :)
